eclair-core/src/main/scala/fr/acinq/eclair/channel/Commitments.scala

```diff
@@ -148,8 +149,9 @@ object Commitments {
     // the HTLC we are about to create is outgoing, but from their point of view it is incoming
     val outgoingHtlcs = reduced.htlcs.filter(_.direction == IN)
 
-    val htlcValueInFlight = UInt64(outgoingHtlcs.map(_.add.amountMsat).sum.toLong)
-    if (htlcValueInFlight > commitments1.remoteParams.maxHtlcValueInFlightMsat) {
+    val htlcValueInFlight = outgoingHtlcs.map(_.add.amountMsat).sum
```

Comment:
I'm afraid this can wrap around since `amountMsat` is a `Long`. If that sum exceeds `Long.MaxValue` it will wrap to a negative value instead of being the correct `UInt64` value we want to use for the comparison, right?